tests/integration/test_charm.py

```diff
@@ -171,10 +128,70 @@ async def test_scale_workload(
     kube_core_client.delete_namespaced_pod(name=leader_pod, namespace=ops_test.model_name)
     await ops_test.model.wait_for_idle(status="active")
 
-    unit_informations = json.loads(
-        (await ops_test.juju("show-unit", app.units[0].name, "--format", "json"))[1]
-    )
-    mattermost_ip = unit_informations[app.units[0].name]["address"]
-    response = requests.get(f"http://{mattermost_ip}:8065", timeout=5)
+    response = requests.get(f"http://{await get_mattermost_ip(ops_test, app)}:8065", timeout=5)
     assert response.status_code == 200
     assert "Mattermost" in response.text
+
+
+async def test_remind_plugin(
+    ops_test: OpsTest,
+    app: Application,
+    test_entities,
+):
+    """
+    arrange: after charm is deployed and ready.
+    act:
+    assert: the plugin should be enabled.
+    """
+
+    plugin_name = "com.github.scottleedavis.mattermost-plugin-remind"
+    cmd = (
+        "MMCTL_LOCAL_SOCKET_PATH=/tmp/mattermost.socket /mattermost/bin/mmctl --local"
+        f" plugin enable {plugin_name}"
+    )
+    await ops_test.juju("run", "--application", app.name, cmd)
+
+    cmd = (
+        "MMCTL_LOCAL_SOCKET_PATH=/tmp/mattermost.socket /mattermost/bin/mmctl --local"
+        " plugin list"
+    )
+    output = await ops_test.juju("run", "--application", app.name, cmd)
+    for line in output[1].splitlines():
+        if line.startswith(plugin_name):
+            assert True, f"{plugin_name} is in enabled plugins."
+            break
+        if line == "Listing disabled plugins":
+            assert False, f"{plugin_name} is not in enabled plugins."
+            break
+
+    mattermost_ip = await get_mattermost_ip(ops_test, app)
+
+    # execute a command
+    reminder = "test01"
+    data = {
+        "channel_id": test_entities["channel"]["id"],
+        "command": f"/remind ~{test_entities['channel']['name']} {reminder} in 1 minute",
+    }
+    requests.post(
+        f"http://{mattermost_ip}:8065/api/v4/commands/execute",
+        data=json.dumps(data),
+        headers=test_entities["headers"],
+        timeout=10,
+    )
+
+    # wait for the reminder to trigger
+    time.sleep(120)
+
+    # execute a command
+    response = requests.get(
+        f"http://{mattermost_ip}:8065/api/v4/channels/{test_entities['channel']['id']}/posts",
+        headers=test_entities["headers"],
+        timeout=10,
+    )
+    posts = response.json()
+    assert any(
+        (
+            x[1]["message"] == f'@test asked me to remind you "{reminder}".'
```

Comment:
Should `x` be called something more meaningful? Also, using `any` in `pytest` means that the failure message isn't great, perhaps consider adding a custom message?

Reply:
Renamed and added a custom message